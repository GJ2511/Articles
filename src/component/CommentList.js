import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

function CommentList({ comments }) {
    const getlist = () => {
        return comments.map((comment, index) => {
            return (
                <li className="media" key={index}>
                    <a href="#" className="float-left">
                        <img src={comment.author.image} alt="" className="rounded-circle" />
                    </a>
                    <div className="media-body">
                        <span className="text-muted float-right">
                            <small className="text-muted">
                                {moment(comment.updatedAt).format('MMMM, Do YYYY, hh:mm a')}
                            </small>
                        </span>
                        <strong className="text-success">@{comment.author.username}</strong>
                        <p>{comment.body}</p>
                    </div>
                </li>
            );
        });
    };

    return <ul className="media-list"> {getlist()}</ul>;
}

CommentList.propTypes = {
    comments: PropTypes.array.isRequired,
};

export default CommentList;
