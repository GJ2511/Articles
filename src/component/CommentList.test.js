import React from 'react';

import CommentList from './CommentList';

const props = {
    authenticated: true,
    comments: [
        {
            id: 1,
            createdAt: '2016-02-18T03:22:56.637Z',
            updatedAt: '2016-02-18T03:22:56.637Z',
            body: 'It takes a Jacobian',
            author: {
                username: 'jake',
                bio: 'I work at statefarm',
                image: 'https://i.stack.imgur.com/xHWG8.jpg',
                following: false,
            },
        },
    ],
    currentUser: {
        email: 'jake@jake.jake',
        token: 'jwt.token.here',
        username: 'jake',
        bio: 'I work at statefarm',
        image: null,
    },
    handleDeleteComment: jest.fn(),
};

describe('<CommentList tests />', () => {
    test('match snapshot', () => {
        expect(shallow(<CommentList {...props} />)).toMatchSnapshot();
    });
});
