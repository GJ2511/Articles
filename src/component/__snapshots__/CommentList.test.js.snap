// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentList tests /> match snapshot 1`] = `
<ul
  className="media-list"
>
   
  <li
    className="media border-light border-bottom p-1"
    key="0"
  >
    <a
      className="float-left mr-1"
      href="#"
    >
      <img
        alt=""
        className="rounded-circle"
        src="https://i.stack.imgur.com/xHWG8.jpg"
        style={
          Object {
            "height": "50px",
          }
        }
      />
    </a>
    <div
      className="media-body"
    >
      <span
        className="text-muted float-right"
      >
        <small
          className="text-muted"
        >
          February, 18th 2016, 08:52 am
        </small>
      </span>
      <strong
        className="text-success"
      >
        @
        jake
      </strong>
      <p>
        It takes a Jacobian
      </p>
      <span
        className="float-right hover-hand"
        onClick={[Function]}
      >
        <svg
          className="bi bi-trash-fill"
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.5 1a1 1 0 0 0-1 1v1a1 1 0 0 0 1 1H3v9a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2V4h.5a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H10a1 1 0 0 0-1-1H7a1 1 0 0 0-1 1H2.5zm3 4a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7a.5.5 0 0 1 .5-.5zM8 5a.5.5 0 0 1 .5.5v7a.5.5 0 0 1-1 0v-7A.5.5 0 0 1 8 5zm3 .5a.5.5 0 0 0-1 0v7a.5.5 0 0 0 1 0v-7z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
  </li>
</ul>
`;
